Faq internals (#70)
reactjs/fr.reactjs.org: content/docs/faq-internals.md

---
id: faq-internals

> title: DOM virtuel et fonctionnalités

Reviewer: ```suggestion
title: DOM virtuel et autres détails
```

Recalage sur `nav.yml`

permalink: docs/faq-internals.html
layout: docs
category: FAQ
---

### Qu'est-ce que le DOM virtuel ? {#what-is-the-virtual-dom}

> Le DOM virtuel (VDOM) est un concept de programmation dans lequel une représentation idéale, ou "virtuelle", d'une interface utilisateur est conservée en mémoire et synchronisée avec le DOM "réel" par une bibliothèque telle que ReactDOM. Ce processus s'appelle [réconciliation](/docs/reconciliation.html).

Reviewer: ```suggestion
Le DOM virtuel (VDOM) est un concept de programmation dans lequel une représentation idéale, ou « virtuelle », d'une interface utilisateur (UI) est conservée en mémoire et synchronisée avec le DOM « réel » par une bibliothèque telle que ReactDOM. Ce processus s'appelle [réconciliation](/docs/reconciliation.html).
```

[Guillemets français](https://github.com/reactjs/fr.reactjs.org/wiki/Meilleures-pratiques-de-traduction#guillemets-français)

> Cette approche active l'API déclarative de React : vous indiquez à React dans quel état, vous souhaitez que l'interface utilisateur se trouve, et il s'assure que le DOM correspond à cet état. Cela extrait la manipulation des attributs, la gestion des événements et la mise à jour manuelle du DOM que sinon vous auriez dû utiliser pour créer votre application.

Reviewer 2: ```suggestion
Cette approche active l'API déclarative de React : vous indiquez à React dans quel état, vous souhaitez que l'interface utilisateur se trouve, et il s'assure que le DOM correspond à cet état. Ça permet de faire abstraction de la manipulation des attributs, de la gestion des événements et de la mise à jour manuelle du DOM que vous auriez dû faire normalement pour créer votre application.
```

> Cette approche active l'API déclarative de React : vous indiquez à React dans quel état, vous souhaitez que l'interface utilisateur se trouve, et il s'assure que le DOM correspond à cet état. Ça permet de faire abstraction de la manipulation des attributs, de la gestion des événements et de la mise à jour manuelle du DOM que vous auriez dû faire normalement pour créer votre application.

Reviewer: ```suggestion
Cette approche rend possible l'API déclarative de React : vous indiquez à React dans quel état, vous souhaitez que l'UI se trouve, et il s'assure que le DOM correspond à cet état. Ça permet de faire abstraction de la manipulation des attributs, de la gestion des événements et de la mise à jour manuelle du DOM que vous auriez dû faire normalement pour créer votre application.
```

Mésinterprétation au début. Sinon, le top 💯

> Puisque le "DOM virtuel" est plus un modèle qu'une technologie spécifique, on l'emploie parfois pour désigner différentes choses. Dans le monde React, le terme "DOM virtuel" est généralement associé aux [éléments React](/docs/rendering-elements.html), car il s'agit des objets représentant l'interface utilisateur. Cependant, React utilise également des objets internes appelés "fibres" pour contenir des informations supplémentaires sur l’arbre des composants. Ils peuvent également être considérés comme faisant partie de l'implémentation du "DOM virtuel" dans React.

Reviewer 2: ```suggestion
Puisque le « DOM virtuel » est plus un modèle qu'une technologie spécifique, on l'emploie parfois pour désigner différentes choses. Dans le monde React, le terme « DOM virtuel » est généralement associé aux [éléments React](/docs/rendering-elements.html), car il s'agit des objets représentant l'interface utilisateur. Cependant, React utilise également des objets internes appelés « fibres » (_fibers_ en anglais, NdT) pour contenir des informations supplémentaires sur l’arbre des composants. Ils peuvent également être considérés comme faisant partie de l'implémentation du « DOM virtuel » dans React.
```

Reviewer 2: Dans la mesure ou la notion de "fibers" est propre à React, on est dans un cas ou ça vaut le coup de laisser une note pour indiquer le mot anglais original.

> Puisque le « DOM virtuel » est plus un modèle qu'une technologie spécifique, on l'emploie parfois pour désigner différentes choses. Dans le monde React, le terme « DOM virtuel » est généralement associé aux [éléments React](/docs/rendering-elements.html), car il s'agit des objets représentant l'interface utilisateur. Cependant, React utilise également des objets internes appelés « fibres » (_fibers_ en anglais, NdT) pour contenir des informations supplémentaires sur l’arbre des composants. Ils peuvent également être considérés comme faisant partie de l'implémentation du « DOM virtuel » dans React.

Reviewer: ```suggestion
Puisque le « DOM virtuel » est plus un modèle qu'une technologie spécifique, on l'emploie parfois pour désigner différentes choses. Dans le monde React, le terme « DOM virtuel » est généralement associé aux [éléments React](/docs/rendering-elements.html), car il s'agit des objets représentant l'interface utilisateur. Cependant, React utilise également des objets internes appelés « fibres » _(fibers, NdT)_ pour conserver des informations supplémentaires sur l’arbre des composants. Ils peuvent également être considérés comme faisant partie de l'implémentation du « DOM virtuel » dans React.
```

> ### Est-ce que le DOM fantôme est le même que le DOM virtuel ? {#is-the-shadow-dom-the-same-as-the-virtual-dom}

Reviewer 2: ```suggestion
### Est-ce que le Shadow DOM est identique au DOM virtuel ? {#is-the-shadow-dom-the-same-as-the-virtual-dom}
```

Reviewer 2: On ne traduit pas _Shadow DOM_ dans le mesure ou c'est une appellation technique spécifique (ça correspond à un concept et une norme technique est est utilisé tel quel en français)

> Non, ils sont différents. Le DOM fantôme est une technologie du navigateur conçue principalement pour parcourir les variables et le CSS des composants Web. Le DOM virtuel est un concept implémenté par les bibliothèques en JavaScript en plus des API de navigateur.

Reviewer 2: ```suggestion
Non, ils sont différents. Le Shadow DOM est une technologie du navigateur conçue principalement pour parcourir les variables et le CSS des composants Web. Le DOM virtuel est un concept implémenté par les bibliothèques en JavaScript en plus des API des navigateurs.
```

Reviewer 2: En général en français, on parles "des navigateurs" quand on veux généraliser le propos plutôt que "du navigateur" au sens conceptuel.

> Non, ils sont différents. Le Shadow DOM est une technologie du navigateur conçue principalement pour parcourir les variables et le CSS des composants Web. Le DOM virtuel est un concept implémenté par les bibliothèques en JavaScript en plus des API des navigateurs.

Reviewer: ```suggestion
Non, ils sont différents. Le Shadow DOM est une technologie du navigateur conçue principalement pour limiter la portée des variables et du CSS dans les Web Components. Le DOM virtuel est un concept implémenté par les bibliothèques en JavaScript en plus des API des navigateurs.
```

- Glissement sémantique sur “scoping”
- On ne traduit pas Web Components (cf. trads déjà validées)

> ### Qu'est-ce que "React Fiber"? {#what-is-react-fiber}

Reviewer: ```suggestion
### Qu'est-ce que « React Fiber » ? {#what-is-react-fiber}
```
